public/API/API_GetUserSetRequests.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ *  API_GetUserSetRequests - gets user's set request list
+ *    u : username
+ *
+ *  array      RequestedSets
+ *    object     [value]
+ *      int        GameID                     unique identifier of the game
+ *      string     Title                      set title
+ *      int        ConsoleID                  the id of the console that the set is for
+ *      string     ConsoleName                the name of the console that the set is for
+ *      string     ImageIcon                  the url of the sets icon
+ *  int        TotalRequests              maximum number of requests a user has
+ *  int        PointsForNext              number of points remaining until maximum request increase
+ */
+
+use App\Models\User;
+use App\Models\UserGameListEntry;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'u' => ['required', 'min:1'],
+]);
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json([], 404);
+}
+
+$requestedSets = UserGameListEntry::with(['game.system'])
+    ->where('user_id', $user->ID)
+    ->where('type', 'achievement_set_request')
+    ->get()
+    ->map(function ($gameData) {
+        return [
+            'GameID' => $gameData->game->id,
+            'Title' => $gameData->game->title,
+            'ConsoleID' => $gameData->game->system->id,
+            'ConsoleName' => $gameData->game->system->name,
+            'ImageIcon' => $gameData->game->ImageIcon,
+        ];
+    });
```

Comment:
Recommend adding an orderBy for consistent results (most important for unit tests, but beneficial for consumers as well). The page is sorted by game title, so that probably makes the most sense, but requires specifying titles in the unit tests as the random game titles wouldn't have a guaranteed order.

Reply:
I was wondering about that, by default it seems to be ordering the result by GameID, they are out of order in my SetRequests table. 

I’ll add the orderBy, I’ll need to make the test return more than one result to get an accurate assert.